### DISCUSSION TITLE
napari ADS plugin for manual segmentation

### DISCUSSION THREAD

**Discussion by akraskovSMDB** [neuropoly/axondeepseg#744] [Help needed] [answered]
Hi, 

I'd like to try napari plugin for manual segmentation. I installed it in a separate conda environment following instructions on napari web site. After installation I can't find ADS plugin in the list of available to install plugins, is this correct?

If yes, I guess it means I need to install it separately. Could you please point me towards the installation instructions? Do I need axondeepseg (and ivadomed) installed in the same conda environment? Do I need it even if I want to use it only for manual segmentation i.e. Load Mask, Fill Axons and Save segmentation (without a need to apply ADS models and Compute morphometrics which I do in batch mode for all images)?

Are you planning to support this plugin in the future or you will be switching to MONAI Label?

Thank you

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @akraskovSMDB !

Your problem is that you installed napari separately - we actually install napari + the plugin in our virtual environment during our installation instructions, and this makes it so the environment is properly "linked" together, which may be the issue you encountered.

I'd recommend trying our installation instructions again (https://axondeepseg.readthedocs.io/en/latest/documentation.html#install-axondeepseg), in particular, note the `pip install -e . plugins/` following the conda env activation which installs both AxonDeepSeg (the current folder, `.`), and the napari plugin (the plugins folder `plugins`).

Afterwards, you can read our section on the napari plugin as well as watch the accompanying video: https://axondeepseg.readthedocs.io/en/latest/documentation.html#graphical-user-interface-gui

**Comment by akraskovSMDB** [comment]
Thank you. Do I have to reinstall axondeepseg? Can I just add napari plugin with full reinstallation of axondeepseg?

**Reply by mathieuboudreau** [reply]
If you already have the latest version of AxonDeepSeg installed, you could try simply `pip install -e . plugins/` from the conda env you created while installng ADS and then running `napari`. If you encounter an issue, I'd recommend reinstalling from scratch just to be sure.

**Reply by mathieuboudreau** [reply]
@akraskovSMDB were you able to get the ADS plugin working on napari?

**Reply by akraskovSMDB** [reply]
Yep, it worked. I created a separate ADS environment for it. Did not want to interfere with my existing one because of GPU setup. 
In my hands, on windows,  in miniconda, following ADS installation instructions does not add GPU support (as confirmed now with a fresh installation),
Previously to add GPU, I had to install pytorch separately, it was a bit painful.
It is a different issue, I will try to napari plugin which works now and I don't need GPU for it.
